src/main/kotlin/org/jitsi/rtp/rtp/RtpPacket.kt

```diff
@@ -72,21 +73,26 @@ open class RtpPacket(
         get() = RtpHeader.getMarker(buffer, offset)
         set(value) = RtpHeader.setMarker(buffer, offset, value)
 
-    var payloadType: Int
-        get() = RtpHeader.getPayloadType(buffer, offset)
-        set(value) = RtpHeader.setPayloadType(buffer, offset, value)
+    /* The four values below (payloadType, sequenceNumber, timestamp, and ssrc)
+     * are very frequently accessed in our pipeline; store their values in
+     * delegated properties, rather than re-reading them from the buffer every time.
+     */
+
+    var payloadType: Int by Delegates.observable(RtpHeader.getPayloadType(buffer, offset)) {
```

Comment:
It may be worth using `observableWhenChanged` (currently in JMT) to avoid calling `RtpHeader.setPayloadType` unnecessarily.

Reply:
Should we move `observableWhenChanged` to jitsi-utils-kotlin, do you think?

Reply:
Ideally, yeah

Reply:
https://github.com/jitsi/jitsi-utils/pull/40